internal/venice-client-common/src/main/java/com/linkedin/venice/schema/merge/SortBasedCollectionFieldOpHandler.java

```diff
@@ -701,6 +702,19 @@ private void sortElementAndTimestampList(
     });
   }
 
+  private void sortElementAndTimestampListInMap(
+      List<ElementAndTimestamp> elementAndTsList,
+      Function<Object, String> keyExtractor) {
+    Comparator<Object> mapKeyComparator = Comparator.comparing(x -> (String) x);
+    elementAndTsList.sort((o1, o2) -> {
```

Comment:
I'm not sure I totally get this.  We're only comparing the keys in the map for the conflict resolution, but if theres a conflict, isn it not a given the keys will be the same?

Reply:
I agree, this seems confusing... shouldn't it be the value which is the tie breaker, rather than the key?

Reply:
Hmm what you are talking about is when comparing old / new value for a same key for one update operation. Here it is different case, basically after we handle all these conflicts for the same keys and we need to sort the active elements by TS. Now two KV pairs can have the same TS inside map data structure and we need to break tie here. So for old logic we were using value to break tie, but I think using key is more lightweight.

Reply:
Basically like this:
After all the conflict resolution, the map could have
{k1=v1, k4=v4, k3=v3, k2=v2, }
put only part length = 1
activeElementsTimestamps = [3, 3, 2]

The last step of the merge update operation is to sort it out so the active elements are sorted by TS. we expect [2, 3, 3] but tie breaking can be either by key or value.

Reply:
ohhhh ok.  We're talking about tie breaking for the sake of SORTING, not conflict resolution?

Reply:
Ok.  synced on chat.  I totally misunderstood what was going on here.  Using key should be fine.  This is ONLY for sorting the result, not conflict resolution.

Reply:
Thanks for clarifying. We should write the DCR spec in the open-source doc.